src/ui/actions/timeline.ts

```diff
@@ -207,45 +219,82 @@ export function seek({
   openSource = false,
   pauseId,
   time,
+  location,
 }: {
   autoPlay?: boolean;
   executionPoint?: ExecutionPoint;
   openSource?: boolean;
   pauseId?: PauseId;
   time: number;
-}): UIThunkAction<void> {
+  location?: Location;
+}): UIThunkAction<Promise<void>> {
   return async (dispatch, getState, { replayClient }) => {
-    // If no ExecutionPoint provided, map time to nearest ExecutionPoint
-    if (executionPoint == null) {
-      time = await clampTime(replayClient, time);
+    dispatch(pauseRequestedAt({ executionPoint, time, location }));
+    dispatch(setTimelineState({ currentTime: time, playback: null }));
+    const focusWindow = replayClient.getCurrentFocusWindow();
 
-      // getPointNearTime could take time while we're processing the recording
-      // so we optimistically set the timeline to the target time
-      dispatch(setTimelineState({ currentTime: time }));
+    if (!pauseId) {
+      // If no ExecutionPoint provided, map time to nearest ExecutionPoint
+      if (executionPoint == null) {
+        time = await clampTime(replayClient, time);
+
+        const nearestEvent = mostRecentPaintOrMouseEvent(time);
+        const timeStampedPoint = await replayClient.getPointNearTime(time);
+        if (
+          nearestEvent &&
+          Math.abs(nearestEvent.time - time) < Math.abs(timeStampedPoint.time - time)
+        ) {
+          executionPoint = nearestEvent.point;
+        } else {
+          executionPoint = timeStampedPoint.point;
+        }
 
-      const nearestEvent = mostRecentPaintOrMouseEvent(time);
-      const timeStampedPoint = await replayClient.getPointNearTime(time);
-      if (
-        nearestEvent &&
-        Math.abs(nearestEvent.time - time) < Math.abs(timeStampedPoint.time - time)
-      ) {
-        executionPoint = nearestEvent.point;
-      } else {
-        executionPoint = timeStampedPoint.point;
+        dispatch(pauseRequestedAt({ executionPoint, time, location }));
+      }
+
+      if (focusWindow === null || !isPointInRegion(executionPoint, focusWindow)) {
+        dispatch(pauseCreationFailed());
+        return;
+      }
+
+      trackEvent("paused");
+
+      try {
+        pauseId = await pauseIdCache.readAsync(replayClient, executionPoint, time);
+      } catch (e) {
+        console.error(e);
+        dispatch(pauseCreationFailed());
+        return;
       }
     }
 
-    assert(executionPoint != null, `Could not find execution point for time ${time}`);
+    assert(executionPoint);
+    dispatch(paused({ executionPoint, time, id: pauseId }));
```

Comment:
Hey look! We're using the `pauseId` param again 🥳

Reply:
Yeah, previously (after my ThreadFront removal) `seek()` would always get the `pauseId` for the given `executionPoint` from `pauseIdCache`. Usually we only have one `pauseId` for any given `executionPoint` (so this shouldn't make a difference), but there is no guarantee for that, so it's good to reconnect the `pauseId` param here.